tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Wcf/WcfCommon.cs

```diff
@@ -94,7 +95,7 @@ internal static Scope CreateScope<TRequestContext>(TRequestContext requestContex
                 Uri requestHeadersTo = requestHeaders.To;
 
                 span.DecorateWebServerSpan(
-                    resourceName: action ?? requestHeadersTo?.LocalPath,
+                    resourceName: string.IsNullOrEmpty(action) ? UriHelpers.GetCleanUriPath(requestHeadersTo?.LocalPath) : action,
```

Comment:
Maybe add a feature flag (enabled by default) and do the obfuscation only when it is set. That will give the users a way to rollback if needed.

Reply:
That sounds like a good idea, I was wondering whether or not if this would be too much of a behavioral breaking change.

Follow up question for this, how exactly are feature flags implemented? Is that in `ConfigurationKeys` and the `FeatureFlags` class or elsewhere?

Reply:
> Follow up question for this, how exactly are feature flags implemented? Is that in ConfigurationKeys and the FeatureFlags class or elsewhere?

Yes

Reply:
> Is that in ConfigurationKeys and the FeatureFlags class or elsewhere

That's the one, and then you need to add a setting to `TracerSettings` and `ImmutableTracerSettings` to flow it through. I would be inclined to call it something like `DD_TRACE_WCF_RESOURCE_OBFUSCATION_ENABLED`, default it true, so they can disable it if it causes issues. 

Please also record the value of that flag in the tracer start up log + telemetry too, so we can track when/if people need to disable it 🙂

Reply:
Okay I _think_ I got the feature flag added, start up log, and telemetry as well:

Feature Flag: c6594cce3
Start up log: bec2bfde6
Telemetry: 0ffe3e843